subsys/bluetooth/audio/ascs.c

```diff
@@ -507,6 +507,10 @@ static void ascs_cp_rsp_add(uint8_t id, uint8_t op, uint8_t code,
 	 */
 	case BT_ASCS_RSP_NOT_SUPPORTED:
 	case BT_ASCS_RSP_TRUNCATED:
+		if (rsp->num_ase) {
+			net_buf_simple_remove_mem(&rsp_buf,
+						  sizeof(*ase_rsp) * rsp->num_ase);
+		}
```

Comment:
The point of this is to remove all previously added responses, so that the buffer only contains a single `bt_ascs_cp_ase_rsp` in the end, right? Or is it no `bt_ascs_cp_ase_rsp`?

I am wondering if it would be simpler to simply recreate the buffer from new in that case, instead of removing the memory here. Performance wise it looks like it'll amount to roughly the same.